Feature/facility

nurgissas opened the pull request:
## What was added
- facility js files
- add images or other context if necessary

## Related Issues
- Post request returns false for URL validation for some values (api/facilities endpoint)
- Get works fine

## Future Works
- What else should be done ?

nurgissas committed:
Add facility files, not up-to-date with 05/13 main merge - just for backup

nurgissas committed:
Add local ignore files to .gitignore

nurgissas committed:
Update facility files

nurgissas committed:
Add addOpeningHours, updateMenu

nurgissas committed:
Remove dotenv from package.json

nurgissas committed:
Debug createFacility and modify updateMenu

cwwojin commented:
Some feedbacks (will add more later)
1. use camelCase for all variables, request fields (keys in body) & all other things in JS except for DB columns
`query='select * from "user" where id = $1', values=[userId]`
2. `"phone", "email"` should be added as fields for facility insert & update
3. do not insert / update `slug` columns as they are generated by DB

nurgissas committed:
Modified variables to have camelCase and add code to show opening_hours and menu when get FacilityById

nurgissas committed:
Add post feature

nurgissas committed:
Debug post-related methods

nurgissas committed:
menu and post are modified, post uses dummy author_id

nurgissas committed:
Post, menu successfully handled, passed unit tests

nurgissas committed:
Added option to createFacility with post parsed in

nurgissas committed:
Add preference option to facility

nurgissas committed:
add preference, stamp reward, stamp ruleset

nurgissas committed:
Fix all camelCases and updated stamp rewards, ruleset -get, bug fixes

nurgissas committed:
Fix overall according to requirements

nurgissas committed:
Fix2 of services

nurgissas committed:
Add delete functionality, and many more

cwwojin committed:
fix : minor edits & remove submodule

- remove submodule
- change imgUri, logoImgUri field validation : remove notEmpty constraint
- change endpoint for GET /stamp-ruleset-rewards -> /stamp-ruleset
- GET facility by id - remove posts from result

cwwojin committed:
Merge branch 'main' into feature/facility

cwwojin committed:
fix : edit package-lock.json

cwwojin commented:
### Changes made
- make imgUri, logoimgUri empty-able (remove notEmpty constraint)
  - will change again when I add image upload feature
- GET stamp ruleset : change path to `/stamp-ruleset-rewards` -> `/stamp-ruleset`
- GET facility by id : remove retrieval of posts, as it can be slower and theres already another request for it
- Remove submodule
- Merge w/ main

let me know if you have any feedbacks!
